src/libsync/propagateupload.cpp

```diff
@@ -894,5 +898,392 @@ void PropagateUploadFile::abortWithError(SyncFileItem::Status status, const QStr
+                QString httpStatus = reader.readElementText();
+                item.insert(name, httpStatus);
+//                if (httpStatus.startsWith("HTTP/1.1 200")) {
+//                    currentPropsHaveHttp200 = true;
+//                } else {
+//                    currentPropsHaveHttp200 = false;
+//                }
+            } else if (name == QLatin1String("prop")) {
+                insideProp = true;
+                continue;
+            } else if (name == QLatin1String("multistatus")) {
+                insideMultiStatus = true;
+                continue;
+            }
+        }
+
+        if (type == QXmlStreamReader::StartElement && insidePropstat && insideProp) {
+            if (name == QLatin1String("oc-path")){
+                currentOcPath = reader.readElementText(QXmlStreamReader::SkipChildElements);
+            }
+            else{
+                item.insert(name, reader.readElementText(QXmlStreamReader::SkipChildElements));
+            }
+        }
+
+        // End elements with DAV:
+        if (type == QXmlStreamReader::EndElement) {
+            if (reader.namespaceUri() == QLatin1String("DAV:")) {
+                if (reader.name() == "response") {
+                    currentHref.clear();
+                } else if (reader.name() == "propstat") {
+                    insidePropstat = false;
+                    if (!currentOcPath.isEmpty()){
+                        responseObjectsProperties.insert(currentOcPath, QMap<QString,QString>(item));
+                    }
+                    currentOcPath.clear();
+                    item.clear();
+                } else if (reader.name() == "prop") {
+                    insideProp = false;
+                }
+            }
+        }
+    }
+
+    if (reader.hasError()) {
+        // XML Parser error? Whatever had been emitted before will come as directoryListingIterated
+        qDebug() << "ERROR" << reader.errorString();
+    }
+
+    foreach(auto item, _bundledFiles) {
+        QString itemFilePath(getRemotePath(item->_file));
+        if (responseObjectsProperties.contains(itemFilePath)){
+            QMap<QString, QString> fileProperties = responseObjectsProperties.value(itemFilePath);
+
+            //verify if it has all required properties
+            if (fileProperties.contains("status") && (fileProperties.value("status")).startsWith("HTTP/1.1 200")){
+                // Check if the file still exists
+                const QString fullFilePath(_propagator->getFilePath(item->_file));
+                if( !FileSystem::fileExists(fullFilePath) ) {
+                    //abortWithError(SyncFileItem::SoftError, tr("The local file was removed during sync."));
+                }
+
+                if (! FileSystem::verifyFileUnchanged(fullFilePath, item->_size, item->_modtime)) {
+                    _propagator->_anotherSyncNeeded = true;
+                }
+
+                //OC-FileID section
```

Comment:
I would say it needs to have fileID, Etag and mtime:accepted else it's an error.. 
@ogoffart ?